postgresql/resource_postgresql_event_trigger.go

```diff
@@ -0,0 +1,408 @@
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			eventTriggerDatabaseAttr: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The database where the event trigger is located. If not specified, the provider default database is used.",
+			},
+			eventTriggerFunctionSchemaAttr: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Schema where the function is located.",
+			},
+			eventTriggerStatusAttr: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "enable",
+				Description: "These configure the firing of event triggers. A disabled trigger is still known to the system, but is not executed when its triggering event occurs",
+				ValidateFunc: validation.StringInSlice([]string{
+					"disable",
+					"enable",
+					"enable_replica",
+					"enable_always",
+				}, false),
+			},
+			eventTriggerOwnerAttr: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The user name of the owner of the event trigger. You can't use 'current_role', 'current_user' or 'session_user' in order to avoid drifts",
+				ValidateFunc: validation.StringNotInSlice([]string{
+					"current_role",
+					"current_user",
+					"session_user",
+				}, true),
+			},
+		},
+	}
+}
+
+func resourcePostgreSQLEventTriggerCreate(db *DBConnection, d *schema.ResourceData) error {
+	eventTriggerName := d.Get(eventTriggerNameAttr).(string)
+	d.SetId(eventTriggerName)
+
+	b := bytes.NewBufferString("CREATE EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerOn := d.Get(eventTriggerOnAttr).(string)
+	fmt.Fprint(b, " ON ", eventTriggerOn)
+
+	if filters, ok := d.GetOk(eventTriggerFilterAttr); ok {
+		filters := filters.([]interface{})
+
+		for i, filter := range filters {
+			filter := filter.(map[string]interface{})
+
+			if variable, ok := filter[eventTriggerFilterVariableAttr]; ok {
+				if i == 0 {
+					fmt.Fprint(b, " WHEN ", variable)
+				} else {
+					fmt.Fprint(b, " AND ", variable)
+				}
+			}
+
+			if values, ok := filter[eventTriggerFilterValueAttr]; ok {
+				var new_values []string
+
+				for _, value := range values.([]interface{}) {
+					new_values = append(new_values, pq.QuoteLiteral(value.(string)))
+				}
+
+				fmt.Fprint(b, " IN (", strings.Join(new_values, ","), ")")
+			}
+		}
+	}
+
+	eventTriggerFunction := d.Get(eventTriggerFunctionAttr).(string)
+	eventTriggerSchema := d.Get(eventTriggerFunctionSchemaAttr).(string)
+	fmt.Fprint(b, " EXECUTE FUNCTION ", pq.QuoteIdentifier(eventTriggerSchema), ".", eventTriggerFunction, "()")
+
+	createSql := b.String()
+
+	// Enable or disable the event trigger
+	b = bytes.NewBufferString("ALTER EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerEnabled := d.Get(eventTriggerStatusAttr).(string)
```

Comment:
Does it work for `enable_replica` and `enable_always` ? 
it seems to generate `ALTER EVENT TRIGGER name enable_replica` where the documentation says:
```
ALTER EVENT TRIGGER name ENABLE [ REPLICA | ALWAYS ]
```

Reply:
I actually checked with your test by adding an extra step like:
```
            {
                Config: fmt.Sprintf(testAccPostgreSQLEventTriggerConfig, dbName, schemas[0], "enable"),
                Check: resource.ComposeTestCheckFunc(
                    testAccCheckPostgresqlEventTriggerExists("postgresql_event_trigger.event_trigger", dbName),
                ),
            },
            {
                Config: fmt.Sprintf(testAccPostgreSQLEventTriggerConfig, dbName, schemas[0], "enable_always"),
                Check: resource.ComposeTestCheckFunc(
                    testAccCheckPostgresqlEventTriggerExists("postgresql_event_trigger.event_trigger", dbName),
                ),
            },
```
(and changing the conf definition to take `%[3]s` instead of "enabled" directly)
so it tries to alter the status of the event and it indeed fails with:
```
Error: pq: syntax error at or near "enable_always"
```

Full PG logs being:
```
ERROR:  syntax error at or near "enable_always" at character 34
STATEMENT:  ALTER EVENT TRIGGER "event_trigger_test" enable_always
```

Reply:
I've added a map to fix this